indico/modules/events/registration/badges.py

```diff
@@ -94,10 +94,10 @@ def _draw_badge(self, canvas, registration, template, tpl_data, pos_x, pos_y):
             if placeholder:
                 if placeholder.group == 'registrant':
                     text = placeholder.render(registration)
-                else:
+                elif placeholder.group == 'event':
                     text = placeholder.render(registration.event)
-            elif item['text']:
-                text = item['text']
+                else:
```

Comment:
I'd use `elif placeholder.group == 'fixed'` here and else fail with an error (since that'd mean there's a placeholder with an unknown/unhandled group)